### DISCUSSION TITLE
Modal/Drawer isLazy prop

### DISCUSSION THREAD

**Discussion by matt-tingen** [chakra-ui/chakra-ui#5569] [Ideas]
In #803, it was decided to make `Popover` non-lazy but keep `Modal` lazy. I believe this predates the `isLazy` prop which was ultimately added to `Popover` to give more control over this behavior. Is there now appetite for also adding an `isLazy` prop to `Modal` (defaulting true to preserve the existing behavior) to allow it to keep its contents mounted?

**Comment by kaspnilsson** [comment]
I would really appreciate this feature.

**Reply by marjan-pluto** [reply]
Some drawers which are expensive to mount can definitely be improved if they are not lazy. @segunadebayo

**Comment by ShafranIvan** [comment]
Workaround in case somebody needs it


```
const { isOpen, onClose, onOpen } = useDisclosure()

return (
  <Modal blockScrollOnMount = { isOpen } isOpen = { true} onClose = { onClose } >
    {isOpen && <ModalOverlay />}
    <Box visibility={isOpen ? 'visible' : 'hidden'}>
      <ModalContent>
        <ModalBody>
          {/* Your content here */}
        </ModalBody>
      </ModalContent>
    </Box>
  </Modal>
)
```

**Reply by DonArtisan** [reply]
Thank you man, you saved my life
